### PR TITLE
add new pattern SERVICE_NAME variable for stat prefix

### PR DESCRIPTION
**Pull request by Ferdudas97**
**Please provide a description of this PR:**

fix: #52177
Add %SERVICE_NAME% variable in statPrefix for inbound and outbound clusters

### CODE REVIEW

File: pilot/pkg/networking/telemetry/telemetry.go [hunk 1]
```diff
@@ -37,6 +38,7 @@ var (
 // BuildStatPrefix builds a stat prefix based on the stat pattern.
 func BuildStatPrefix(statPattern string, host string, subset string, port *model.Port, targetPort int, attributes *model.ServiceAttributes) string {
 	prefix := strings.ReplaceAll(statPattern, serviceStatPattern, shortHostName(host, attributes))
+	prefix = strings.ReplaceAll(prefix, serviceNameStatPattern, attributes.Name)
```
Comment (ramaraochavali) [inline pilot/pkg/networking/telemetry/telemetry.go:41]: attributes only exist for Kubernetes. Please check how shortHostName is implemented

Comment (Ferdudas97) [reply pilot/pkg/networking/telemetry/telemetry.go:41]: should i put there a hostname when there is no attributes?

Comment (ramaraochavali) [reply pilot/pkg/networking/telemetry/telemetry.go:41]: yes

**Comment by zirain** [comment]
should also update the doc

File: pilot/pkg/networking/telemetry/telemetry.go (hunk 1 above)
Comment (Ferdudas97) [reply pilot/pkg/networking/telemetry/telemetry.go:41]: Imo is better to replace it with empty string as it is with subset

Comment (ramaraochavali) [reply pilot/pkg/networking/telemetry/telemetry.go:41]: name is equivalent to host for non k8s services like service entries

Comment (Ferdudas97) [reply pilot/pkg/networking/telemetry/telemetry.go:41]: ok

**Comment by Ferdudas97** [comment]
> should also update the doc

https://github.com/istio/istio.io/pull/15449

**Comment by zirain** [comment]
> > should also update the doc
> 
> [istio/istio.io#15449](https://github.com/istio/istio.io/pull/15449)

it's not the right way, you should update api, then automator will do the rest.

**Comment by howardjohn** [comment]
LGTM, will leave to @ramaraochavali as they are expert on this

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**